src/cow-react/modules/limitOrders/hooks/useCalculateRate.ts

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from 'react'
+import { useAtomValue } from 'jotai'
+
+import { limitOrdersAtom } from '../state/limitOrdersAtom'
+import { limitRateAtom } from '../state/limitRateAtom'
+import { limitDecimals } from '../utils/limitDecimals'
+
+// Applies rate to provided value which can be INPUT or OUTPUT
+export function useCalculateRate() {
+  const { inputCurrencyAmount, outputCurrencyAmount } = useAtomValue(limitOrdersAtom)
+  const { activeRate, isInversed } = useAtomValue(limitRateAtom)
+
+  return useCallback(() => {
```

Comment:
There are no parameters in callback, can we use useMemo instead?

Reply:
We could but then the calculation is called on every hook dependency change which is unnecessary so I would rather keep it as a function for now.